src/EventListener/EndEmptySessionSubscriber.php

```diff
@@ -22,13 +23,11 @@ public function onKernelResponse(FilterResponseEvent $event)
         $request = $event->getRequest();
         $session = $request->getSession();
 
-        if (!$session || (!$session->isStarted() && !$request->hasPreviousSession())) {
+        if (!$session instanceof Session || !$session->isStarted() || !$session->isEmpty()) {
```

Comment:
`$session->isEmpty()` is `@internal`, but no other way to check that it is empty.